src/libYARP_sig/include/yarp/sig/Image.h

```diff
@@ -173,7 +206,7 @@ class YARP_sig_API yarp::sig::Image : public yarp::os::Portable {
      * @return address of pixel in memory
      */
     inline unsigned char *getPixelAddress(int x, int y) const {
-        return (unsigned char *)(data[y] + x*imgPixelSize);
+        return (unsigned char *)(data[y] + x*getPixelSize());
```

Comment:
Sorry, but I'm a bit afraid that not having this as a variable and calling this function instead, might have a heavy impact on user code performances... Just image a for loop calling `getPixelAddress()` for each pixel!

I'd rather restore the `imgPixelSize` variable and use it as a cache whenever the PixelCode is changed